addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/ExitStatusJobUnitTest.java

```diff
@@ -348,22 +349,93 @@ void shouldSetExitCode(String alertrisk, String exitcode) {
+        job.getParameters().setErrorExitValue(4);
+        job.getParameters().setErrorLevel("high");
+        job.getParameters().setWarnLevel("medium");
+        job.verifyParameters(progress);
+        job.runJob(new AutomationEnvironment(progress), progress);
+
+        // Then
+        assertThat(ExtensionAutomation.getExitOverride(), is(equalTo(Integer.parseInt(exitcode))));
+    }
+
+    @Test
+    void shouldSetExitCodeBasedOnAlerts() {
+        // Given
+        ExitStatusJob job = new ExitStatusJob();
+        AutomationProgress progress = new AutomationProgress();
+        progress.addJobResultData(getTestData("medium", "false positive", true, true));
+        // When
+        job.getParameters().setOkExitValue(1);
+        job.getParameters().setWarnExitValue(2);
+        job.getParameters().setErrorExitValue(3);
+        job.getParameters().setErrorLevel("high");
+        job.getParameters().setWarnLevel("medium");
+        job.verifyParameters(progress);
+        job.runJob(new AutomationEnvironment(progress), progress);
+        // Then
+        assertThat(ExtensionAutomation.getExitOverride(), is(equalTo(3)));
+    }
 
-        JobResultData data =
-                new JobResultData("test") {
+    private static JobResultData getTestData(String alertLevel) {
+        return getTestData(alertLevel, "Medium", true, false);
+    }
 
-                    @Override
-                    public String getKey() {
-                        return "test";
-                    }
+    private static JobResultData getTestData(
+            String alertLevel,
+            String confidence,
+            boolean includeBaseAlert,
+            boolean includeExtraAlerts) {
+        Alert alert =
+                new Alert(
+                        -1,
+                        JobUtils.parseAlertRisk(alertLevel),
+                        JobUtils.parseAlertConfidence(confidence),
+                        "test");
+        List<Alert> extraAlerts = getAlerts();
+
+        return new JobResultData("test") {
+
+            @Override
+            public String getKey() {
+                return "test";
+            }
+
+            @Override
+            public Collection<Alert> getAllAlertData() {
+                List<Alert> alertList = new ArrayList<>();
+                if (includeBaseAlert) {
+                    alertList.add(alert);
+                }
+                if (includeExtraAlerts) {
+                    alertList.addAll(extraAlerts);
+                }
+                return alertList;
+            }
+        };
+    }
 
-                    @Override
-                    public Collection<Alert> getAllAlertData() {
-                        return Arrays.asList(alert);
-                    }
-                };
-        return data;
+    private static List<Alert> getAlerts() {
+        Alert otherAlert = new Alert(-1, 3, 3, "test");
+        return List.of(otherAlert, otherAlert, otherAlert, otherAlert);
     }
```

Comment:
This should be inlined, also why duplicate them? One more than the FP is enough.

Reply:
Tweaked

Reply:
Sorry I was taking your comment much more isolated/specifically. Tweaked again 😀